lib/child_process.js

```diff
@@ -418,13 +464,34 @@ function spawnSync(/*file, args, options*/) {
 
   debug('spawnSync', opts.args, options);
 
+  // Validate the timeout, if present.
+  if (options.timeout != null &&
+      !(typeof options.timeout === 'number' && options.timeout >= 0)) {
+    throw new TypeError('"timeout" must be a non-negative number');
+  }
+
+  // Validate maxBuffer, if present.
+  if (options.maxBuffer != null &&
+      !(typeof options.maxBuffer === 'number' &&
+      (options.maxBuffer === options.maxBuffer >>> 0))) {
+    throw new TypeError('"maxBuffer" must be an unsigned integer');
```

Comment:
Why say 'unsigned integer' here when other error messages say `non-negative number`. Would prefer these to be consistent.